perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ExtSourcesManagerBlImpl.java

```diff
@@ -398,6 +398,9 @@ public Candidate getCandidate(PerunSession perunSession, Map<String,String> subj
 		// Filter attributes
 		Map<String, String> attributes = new HashMap<>();
 		for (String attrName: subjectData.keySet()) {
+			if (attrName == null) {
```

Comment:
I'm not sure, why is this change part of the pull-request, since its not related to PERUNOBSERVER role at all. Does it fixes some tests maybe ? I'm not saying, that change itself is wrong, but please clarify it. Thank you.

Reply:
I am sorry, somehow I managed to leave changes from my testing branch in perunobserver, I'll get rid of it.